tests/test_config.py

```diff
@@ -175,23 +204,64 @@ def testHierarchy(self):
         # Check we get the same top level keys
         self.assertEqual(set(c.names(topLevelOnly=True)), set(c.data.keys()))
 
+        # Check that we can iterate through items
+        for k, v in c.items():
+            self.assertEqual(c[k], v)
+
         # Check that lists still work even if assigned a dict
         c = Config({"cls": "lsst.daf.butler",
+                    "formatters": {"calexp.wcs": "{component}",
+                                   "calexp": "{datasetType}"},
                     "datastores": [{"datastore": {"cls": "datastore1"}},
                                    {"datastore": {"cls": "datastore2"}}]})
-        c["datastores.1.datastore"] = {"cls": "datastore2modified"}
-        self.assertEqual(c["datastores.0.datastore.cls"], "datastore1")
-        self.assertEqual(c["datastores.1.datastore.cls"], "datastore2modified")
+        c[".datastores.1.datastore"] = {"cls": "datastore2modified"}
+        self.assertEqual(c[".datastores.0.datastore.cls"], "datastore1")
+        self.assertEqual(c[".datastores.1.datastore.cls"], "datastore2modified")
         self.assertIsInstance(c["datastores"], collections.Sequence)
 
         # Test that we can get all the listed names.
         # and also that they are marked as "in" the Config
-        # Two distinct loops
-        for n in c.names():
+        # Try delimited names and tuples
+        for n in itertools.chain(c.names(), c.nameTuples()):
             val = c[n]
             self.assertIsNotNone(val)
             self.assertIn(n, c)
 
+        names = c.names()
+        nameTuples = c.nameTuples()
+        self.assertEqual(len(names), len(nameTuples))
+        self.assertGreater(len(names), 5)
```

Comment:
Why `assertGreater` here?  Don't you know what the length should be?

Reply:
Fixed to use Equal.